### PR TITLE
chore: prevent weird behaviour of pre-commit hook

### LINKED ISSUE #956: 🐞[infra]: weird behaviour by commit hook
**Describe the bug**
After commit, the code gets automatically formatted and linting throws an error in weird characters.

**What is the current behavior?**
Error and bad linting.

**To Reproduce**
1. I changed a few .ts files and after doing `git commit`
2. FIles gets automatically formatted
3. Running `yarn lint` throws error because of bad auto linting.

**Expected behavior**
No formatting after commit 

**Screenshots**

Screenshot from 2019-06-11 14-57-02

### PR DESCRIPTION
**Pull request by jamesgeorge007**
**What kind of change does this PR introduce?**

bugfix

**Did you add tests for your changes?**
N/A

**If relevant, did you update the documentation?**
N/A

**Summary**
Fixed the weird behavior of pre-commit hook as mentioned in #956

Fixes #956 

**Does this PR introduce a breaking change?**

Nope

**Other information**
N/A

### CODE REVIEW

**Comment by jamesgeorge007** [comment]
cc @evenstensberg

**Comment by jamesgeorge007** [comment]
@ematipico Are these rules to be included within .`eslintrc.js` ?

```js
{
  "extends": [
    "plugin:@typescript-eslint/recommended",
    "prettier",
    "prettier/@typescript-eslint"
  ]
}
```

Is this the required workaround :thinking:

**Comment by jamesgeorge007** [comment]
@ematipico Are these rules to be included within .`eslintrc.js`?

```js
{
  "extends": [
    "plugin:@typescript-eslint/recommended",
    "prettier",
    "prettier/@typescript-eslint"
  ]
}
```

Is this the required workaround :thinking:

**Comment by ematipico** [comment]
We have to include this in all the `.eslintrc` files we have inside the repository. It's not a workaround, it's correct integration between `@typescript-eslint` and `prettier`

File: package.json [hunk 1]
```diff
@@ -34,8 +34,8 @@
     "commit": "git-cz",
     "docs": "typedoc",
     "format": "npm run format:js && npm run format:ts",
-    "format:js": "prettier-eslint ./bin/*.js ./bin/**/*.js ./test/**/*.js ./packages/**/**/*.js ./packages/**/*.js --write",
```
Comment (anshumanv) [inline package.json]: I think this should stay because this intends to format files.

**Comment by jamesgeorge007** [comment]
@ematipico All instances of `.eslintrc` within `packages` directory is found to have the respective config:-  

```json
{
	"root": true,
	"extends": [
		"../../.eslintrc.js",
		"plugin:@typescript-eslint/recommended",
		"prettier",
		"prettier/@typescript-eslint"
	],
	"parser": "@typescript-eslint/parser",
	"plugins": ["@typescript-eslint"]
}
```

How should it be refactored :think

**Comment by jamesgeorge007** [comment]
cc @ematipico

**Comment by jamesgeorge007** [comment]
cc @ematipico

**Comment by jamesgeorge007** [comment]
@rishabh3112 CI was failing without those lock file changes.

**Comment by jamesgeorge007** [comment]
cc @ematipico

**Pull request merged**